### ISSUE TITLE
[BUG]  Add error handling around System.ConsolePal.Clear()

### ISSUE THREAD

**Issue by Max-Herbold** [MCCTeam/Minecraft-Console-Client#2333]
### Prerequisites

- [X] I made sure I am running the latest [development build](https://github.com/MCCTeam/Minecraft-Console-Client/releases/latest)
- [X] I tried to [look for similar issues](https://github.com/MCCTeam/Minecraft-Console-Client/issues?q=is%3Aissue) before opening a new one
- [x] I have set `debugmessages=true` in config to diagnose my issue
- [X] I have redacted session tokens and passwords before attaching screenshots

### Minecraft Version

irrelevant

### Console Client Version

Latest

### Expected Behavior

Should not clear an object that cannot be cleared. Add some error handling?

### Actual Behavior

Crashes because cannot clear.

### Steps to Reproduce the bug

1.  Set stdout pipe to some object that does not have a terminal output/cannot be cleared.

### Attach screenshot here (If applicable)

![image](https://user-images.githubusercontent.com/49804267/200427494-3ff30a17-111d-430d-99a9-674c8c80329d.png)

### Operating System

Windows

**Comment by Max-Herbold** [comment]
The solution to this, after going through the source is to use the "BasicIO-NoColor" argument. Add this to docs?

This reverts to just writing, rather than trying to use clean, setcursorpos, and other functions that cannot be applied to an object without a display.

**Comment by Max-Herbold** [comment]
I added this snippet:
`
            TextReader reader;
            reader = Console.In;
            int val;
            Console.Write("Started the thread");
            while (!token.IsCancellationRequested) {
                val = reader.Read();
                if (val != -1) {
                    MessageReceived?.Invoke(null, (char)val + reader.ReadLine());
                }
            }
            Console.WriteLine("Exiting thread");

            return;
`
to the top of the `KeyListener` function and this works properly with stdin. Now I realize, this library is not actually part of the MCC. I will use this custom built binary going forward I wonder if anyone would be interested in having this program operate with pipes working correctly. I'd imagine the best way to do this would if BasicIO is set use the code above ^ otherwise if default use the previous implementation. @breadbyte

**Comment by Max-Herbold** [comment]
Also, the newer `public bool SendChatCommand(string command, PlayerKeyPair? playerKeyPair)` doesn't work with this, I don't know why. I'm using this on 1.18 so just removed it completely.

**Comment by breadbyte** [comment]
> this library is not actually part of the MCC

Technically, from a library standpoint, yes, it's separate from MCC, but I wrote ConsoleInteractive for the sole purpose of using it in MCC.

I separated it with the idea of modularization alongside the .NET 5 port, as terminal handling broke there, and MCC itself shouldn't be deeply concerned with the specifics of the current terminal or stdin/out implementation. (and mainly because the terminal handling code is massive and would pollute the code base)

That being said, I'm currently working on an implementation that will work with headless usage in general (the primary use case was Docker.)

Thank you for the bug report, and I will look into implementing your suggestion.
